jsoniter-scala-benchmark/shared/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/benchmark/Smithy4sCodecs.scala

```diff
@@ -0,0 +1,117 @@
+package com.github.plokhotnyuk.jsoniter_scala.benchmark
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import smithy4s.http.json._
+import smithy4s.Schema
+import smithy4s.api.Discriminated
+import smithy4s.schema.Schema._
+
+object Smithy4sCodecs {
+  val exceptionWithoutDumpConfig: ReaderConfig = ReaderConfig.withAppendHexDumpToParseException(false)
+  val exceptionWithStacktraceConfig: ReaderConfig = ReaderConfig.withThrowReaderExceptionWithStackTrace(true)
+  val tooLongStringConfig: ReaderConfig = ReaderConfig.withPreferredCharBufSize(1024 * 1024)
+  val escapingConfig: WriterConfig = WriterConfig.withEscapeUnicode(true)
+  val prettyConfig: WriterConfig = WriterConfig.withIndentionStep(2).withPreferredBufSize(32768)
+  implicit val adtSchema: Schema[ADTBase] = recursive {
+    val xAlt = struct(int.required[X]("a", _.a))(X.apply).oneOf[ADTBase]("X")
+    val yAlt = struct(string.required[Y]("b", _.b))(Y.apply).oneOf[ADTBase]("Y")
+    val zAlt = struct(adtSchema.required[Z]("l", _.l), adtSchema.required[Z]("r", _.r))(Z.apply).oneOf[ADTBase]("Z")
+    union(xAlt, yAlt, zAlt) {
+      case x: X => xAlt(x)
+      case y: Y => yAlt(y)
+      case z: Z => zAlt(z)
+    }.addHints(Discriminated("type"))
+  }
+  implicit val adtJCodec: JCodec[ADTBase] = JCodec.deriveJCodecFromSchema(adtSchema)
+  val anyValsSchema: Schema[AnyVals] =
+    struct(
+      byte.required[AnyVals]("b", _.b.a).addHints(smithy.api.Required()),
```

Comment:
the `Required` hint is actually not needed, as the information it carries is already captured statically in `.required` (and its absence is captured by `.optional`). 

the smithy4s code generator generically renders values for all hints that are present in the spec, which is why you can see the `addHints(smithy4s.api.Required())` in the generated code, but in fact it has zero incidence. 

So in other words, you can remove `.addHints(smithy.api.Required())` everywhere

Reply:
Thank you, Olivier! Fixed in https://github.com/plokhotnyuk/jsoniter-scala/pull/903/commits/0fdfbb640c71edae3e55254b6b8fa2ca31a5a456